pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SubnetType describes the subnet type.
+type SubnetType string
+
+const (
+	SubnetTypePublic  SubnetType = "public"
+	SubnetTypePrivate SubnetType = "private"
+)
+
+// VPCNetworkConfigurationSpec defines the desired state of VPCNetworkConfiguration
+type VPCNetworkConfigurationSpec struct {
+	// Namespaces to be applied
+	AppliedToNamespaces []string `json:"appliedToNamespaces,omitempty"`
+	// Tier0 gateway or Tier0 VRF path
+	Tier0Gateway string `json:"tier0Gateway,omitempty"`
+	// Edge cluster paths on which the networking elements will be created
+	// +kubebuilder:validation:Required
+	EdgeCluster string `json:"edgeCluster,omitempty"`
+	// NSX-T Project of the namespace associated with
+	NSXTProject string `json:"nsxtProject,omitempty"`
+	// IP Block paths used to allocate public Subnet
+	// +kubebuilder:validation:MinItems=0
+	// +kubebuilder:validation:MaxItems=5
+	PublicIPV4Pools []string `json:"publicIPV4Pools,omitempty"`
+	// Private CIDRs used to allocate private Subnet
+	// +kubebuilder:validation:MinItems=0
+	// +kubebuilder:validation:MaxItems=5
```

Comment:
In my understanding one NSX IPBlock can have multiple CIDRs? And here we plan to define a single IPBlock for all CIDRs, or one for each?

Reply:
We plan to define a single IPBlock for each CIDR.
Checked NSX API, when create IPBlock, only a CIDR can be specified.

PATCH https://<policy-mgr>/policy/api/v1/infra/ip-blocks/Finance-block
{
  "display_name": "Finance-block",
  "description": "Finance-block Description",
  "cidr": "192.168.0.0/16"
}

Reply:
Ok.